nodejs/src/apis/RoomService.ts

```diff
@@ -1,103 +1,111 @@
-    const payload: Record<string, any> = {
-      name: config.name,
-      description: config.description,
-      template_id: config.templateId,
-      region: config.region,
-    };
-    return this.apiService.post(this.basePath, payload);
-  }
+  /**
+   *
+   * @param before Timestamp before which the Rooms were created
+   * @param after Timestamp after which the Rooms were created
+   * @returns a `QueryResultsIterator<HMSRoom>` object
+   */
+  getRoomsByTimeRangeIterable(before?: Date, after?: Date): QueryResultsIterator<HMSRoom> {
+    const timeQueryParams: Record<string, Date> = {};
+    if (before) timeQueryParams["before"] = before;
+    if (after) timeQueryParams["after"] = after;
 
-  async startHLS(config: StartRoomHLSConfig) {
-    const payload: Record<string, any> = {
-      room_id: config.roomId,
-      variants: [{ meeting_url: config.meetingUrl }],
-    };
-    if (config.recording) {
-      payload.recording = {
-        hls_vod: !!config.recording.hlsVod,
-        single_file_per_layer: !!config.recording.singleFilePerLayer,
-      };
-    }
-    if (config.scheduleAt) {
-      payload.schedule_at = Math.floor(config.scheduleAt.getTime() / 1000);
-    }
-    logger.info("starting hls - ", payload);
-    return this.apiService.post("/meetings/hls/start", payload);
+    const queryResultsIterable = new QueryResultsIterator<HMSRoom>(
+      (queryParams: Record<string, any>) => this.apiService.get(this.basePath, queryParams),
+      timeQueryParams
+    );
+    return queryResultsIterable;
   }
 
-  async stopHLS({ roomId }: { roomId: string }) {
-    logger.info("stopping hls for room - ", roomId);
-    await this.apiService.post("/meetings/hls/stop", { room_id: roomId });
+  /**
+   *
+   * @param config Config of the Room to be created
+   * @returns a `HMSRoom` object
+   */
+  async createRoom(config?: HMSCreateRoomConfig): Promise<HMSRoom> {
+    return this.apiService.post(this.basePath, config);
   }
-}
 
-export interface CreateRoomConfig {
-  name?: string;
-  description?: string;
-  templateId?: string;
-  region?: string;
-}
-
-export interface StartRoomHLSConfig {
-  roomId: string;
-  meetingUrl: string;
-  recording?: HLSRecordingConfig;
-  scheduleAt?: Date;
-}
-
-export interface HLSRecordingConfig {
-  hlsVod?: boolean;
-  singleFilePerLayer?: boolean;
-}
+  /**
+   *
+   * @param roomId Room ID
+   * @param options Options of the Room to be updated
+   * @returns a `HMSRoom` object
+   */
+  async updateRoom(roomId: string, options: HMSUpdateRoomOptions): Promise<HMSRoom> {
+    return this.apiService.post(`${this.basePath}/${roomId}`, options);
+  }
 
-export interface HLSRoomState {
-  running: boolean;
-  url?: string;
-  recording?: HLSRecordingConfig;
+  /**
+   *
+   * @param roomId Room ID
+   * @param enabled Enabled status of Room
+   * @returns a `HMSRoom` object
+   */
+  async enableOrDisableRoom(roomId: string, enabled: boolean): Promise<HMSRoom> {
```

Comment:
tsdoc - "disabling a room would block any future attempts of joining that room till its enabled back"